File: src/main/java/kdh/boardproject/dto/category/CategoryListDto.java
package kdh.boardproject.dto.category;

import kdh.boardproject.entity.Category;
import lombok.*;

import java.time.LocalDateTime;

@Data
public class CategoryListDto {
    private Long categoryIdx;
    private String categoryName;
    private String description;

    private LocalDateTime createdAt;
    public CategoryListDto(Category category){
        this.categoryIdx = category.getIdx();
        this.categoryName = category.getCategoryName();
        this.description = category.getDescription();
        this.createdAt = category.getCreatedAt();
    }
}

File: src/main/java/kdh/boardproject/dto/category/CategoryListDto.kt
package kdh.boardproject.dto.category

import kdh.boardproject.entity.Category
import lombok.Data
import java.time.LocalDateTime

@Data
class CategoryListDto(category: Category) {
    var categoryIdx: Long?
    var categoryName: String
    var description: String
    var createdAt: LocalDateTime?

    init {
        categoryIdx = category.idx
        categoryName = category.categoryName
        description = category.description
        createdAt = category.createdAt
    }
}